sretoolbox/container/image.py

```diff
@@ -419,11 +427,12 @@ def _get_auth(self, www_auth):
         for key, value in www_auth.items():
             url += f'{key}={value}&'
 
-        response = requests.get(url, auth=self.auth)
+        get = self.session.get if self.session else requests.get
```

Comment:
I don't know the meaning of the true value check for requests.sessions.Session. I'd expect it to be inherited from "object", so this will work, but that's not what you want to test really. To be safe, please check for `not None`:
```python
get = self.session.get if self.session is not None else requests.get
```

Reply:
right, I really miss null coalescing feature from other langs like `self.session?.get ?? requests.get`

Reply:
Yeah, as a side note, those ternary expressions are not very nice to read... I'd stick to the good old unested if/else.

Reply:
ok, I'll extract a method to make it explicit

Reply:
fixed in https://github.com/app-sre/sretoolbox/pull/105